lib/phobos_db_checkpoint/handler.rb

```diff
@@ -31,7 +45,15 @@ def around_consume(payload, metadata)
           end
 
           event_action = instrument('db_checkpoint.event_action', event_metadata) do
-            yield
+            begin
```

Comment:
I like @tulios solution. It keeps the old behaviour and the example @klippx brought is very clean.

TL;DR - it boils down to which of the following costs more:
* consuming out-of order, but increasing throughput
* ensuring the order, but stoping the process

For me it does make sense to block a kafka partition on errors by default. Remember that blocking a Phobos handler forever actually means blocking one partition, not the entire process. Since the main point for many Kafka usages is that order is guaranteed by partition, if my consumer marks a message as failed, continues the process and then enters an unexpected state because we consumed messages out of order this might cause really huge losses, where a blocked consumer will likely cause frustration.